changelog/12135.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix subfixtures adding their finalizer multiple times to parent fixtures, causing incorrect teardown ordering in some instances.
```

Comment:
"a fixture that this fixture requests" turned out to be very wordy and hard to read, so I swapped to using "dependent fixtures" - which seems to be somewhat established and shouldn't be possible to interpret as the reverse.

Switched to non-intuitive + unreliable (since teardown order will change depending on which fixture was last requested)